SCAutolib/cli_commands.py

```diff
@@ -29,35 +64,13 @@ def cli(ctx, force, verbose, conf):
     logger.setLevel(verbose)
     ctx.ensure_object(dict)  # Create a dict to store the context
     ctx.obj["FORCE"] = force  # Store the force option in the context
-    ctx.obj["CONTROLLER"] = Controller(conf)
-
-
-@click.command()
-@click.option("--ca-type", "-t",
-              required=False,
-              default='all',
-              type=click.Choice(['all', 'local', 'ipa'], case_sensitive=False),
-              show_default=True,
-              help="Type of the CA to be configured. If not set, all CA's "
-                   "from the config file would be configured")
-@click.pass_context
-def setup_ca(ctx, ca_type):
-    """
-    Configure the CA's in the config file. If more than one CA is
-    specified, specified CA type would be configured.
-    """
-    cnt = ctx.obj["CONTROLLER"]
-    if ca_type == 'all':
-        cnt.setup_local_ca(force=ctx.obj["FORCE"])
-        cnt.setup_ipa_client(force=ctx.obj["FORCE"])
-    elif ca_type == 'local':
-        cnt.setup_local_ca(force=ctx.obj["FORCE"])
-    elif ca_type == 'ipa':
-        cnt.setup_ipa_client(force=ctx.obj["FORCE"])
-    exit(ReturnCode.SUCCESS.value)
+    parsed_conf = None
+    if ctx.invoked_subcommand and not gui:
```

Comment:
This should be something like `if ctx.invoked_subcommand != gui:` otherwise it is not working when I am using the prepare subcommand and not the gui subcommand.

Reply:
I changed this and I added a --no-conf option (we can change the name if that doesn't make sense) to completely skip reading a config.   When I tried running without that, I hit issues because the controller still tried to read the file even though the path check was moved and skipped.   Let me know if you think there's a better way to handle this.

Reply:
TBH I don't really like this --no-conf flag, I have modified (supposedly) the controller to accept None config and skip trying to read it. Can you provide some more data on what fails, when I run the code it worked just fine.

Reply:
Yes, let me drop that and I'll send some output for what I'm seeing.

Reply:
Ohh yeah I am so sorry, it is suppose to be a string and not the function object!